File: core/src/main/java/com/mcreater/amclcore/command/script/CmdletScriptProvider.java
package com.mcreater.amclcore.command.script;

import com.mcreater.amclcore.command.CommandArg;

import java.util.List;
import java.util.stream.Collectors;

public class CmdletScriptProvider implements ScriptProvider {
    public static final CmdletScriptProvider INSTANCE = new CmdletScriptProvider();

    private CmdletScriptProvider() {
    }

    public String toScript(List<CommandArg> args) {
        return args.stream()
                .map(CommandArg::toString)
                .map(s -> s.contains(" ") ? ("\"" + s + "\"") : s)
                .collect(Collectors.joining(" "));
    }

    public String getFileExtension() {
        return ".bat";
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/script/CmdletScriptProvider.kt
package com.mcreater.amclcore.command.script

import com.mcreater.amclcore.command.CommandArg
import java.util.stream.Collectors


class CmdletScriptProvider private constructor() : ScriptProvider {
    override fun toScript(args: List<CommandArg?>?): String? {
        return args?.stream()
            ?.map { it.toString() }
            ?.map { if (it.contains(" ")) "\"" + it + "\"" else it }
            ?.collect(Collectors.joining(" "))
    }

    override val fileExtension: String
        get() = ".bat"

    companion object {
        val INSTANCE = CmdletScriptProvider()
    }
}

